src/AudioSystem.cpp

```diff
@@ -1,14 +1,27 @@
 #include <iostream>
+#include <utility>
+#include <vector>
 
 #include "AudioSystem.h"
 
 #include "SDL.h"
 #include "SDL_mixer.h"
 
 namespace {
-    const int MIX_FREQ_HZ = 44100; // Sound sampling frequency in Hz. Change to "22050" if too slow.
-    const int MIX_NUM_CHANNELS = 2; // Number of sound channels for output
-    const int MIX_CHUNK_SIZE = 4096; // Bytes used per output sample
+    // Sound sampling frequency in Hz. Change to "22050" if too slow.
+    const int MIX_FREQ_HZ = 44100;
+    // Number of sound channels for output
+    const int MIX_NUM_CHANNELS = 2;
+    // Bytes used per output sample
+    const int MIX_CHUNK_SIZE = 4096;
+    // Information for loading sound assets. First value is the key, second is the path.
+    const std::vector<std::pair<const char*, const char*>> SOUND_ASSETS_INFO = {
+        {"beat", "assets/audio/beat.wav"}
```

Comment:
An enum for the key compared to a string makes this a lot safer.

Reply:
So you want an enum for each and every type of audio asset? That would work if we only have a few handful assets but once we start adding in sound effects for every little thing, that might get out of hand real quick

Reply:
We're going to hardcoding every string anyways. I'd rather keep that huge enum, makes it compiler errors if we spell it wrong (and IDE completions), strings make hard to debug runtime errors.

Reply:
That's fair. I'll get that done